src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -25,76 +24,69 @@ export type CheckboxInputProps = CheckboxHTMLElementProps & {
   id: string;
 };
 
-const checkboxIconNameMap = {
-  indeterminate: 'checkbox-indeterminate',
-  true: 'checkbox-checked',
-  false: 'checkbox-unchecked',
-};
-const CheckboxSvg = ({
-  checked,
-  disabled,
-}: {
-  checked: boolean | 'indeterminate';
-  disabled?: boolean;
-}) => {
-  const iconClassName = clsx(
-    styles['checkbox__icon'],
-    disabled && styles['checkbox__icon--disabled'],
-  );
-  return (
-    <Icon
-      className={iconClassName}
-      name={
-        checkboxIconNameMap[`${checked}`] as
-          | 'checkbox-indeterminate'
-          | 'checkbox-checked'
-          | 'checkbox-unchecked'
-      }
-      purpose="decorative"
-      size="1.5rem"
-    />
-  );
-};
-
 /**
  * Checkbox input element, exported for greater flexibility.
  * You must provide an `id` prop and connect it to a visible label.
  */
 export const CheckboxInput = React.forwardRef<
   HTMLInputElement,
   CheckboxInputProps
->(({ checked = false, className, disabled, ...other }, ref) => {
```

Comment:
Removing the default prop value lets the component be uncontrolled (because `value` is `undefined`).